content/courses/native-onchain-development/cross-program-invocations.md

```diff
@@ -797,19 +809,80 @@ pub fn initialize_token_mint(program_id: &Pubkey, accounts: &[AccountInfo]) -> P
 }
 ```
 
-#### 7. Build and deploy
+### 7. Build and Deploy
+
+Now we're ready to build and deploy our program! You can build the program by
+running `cargo build-sbf`.
+
+```sh
+cargo build-sbf
+```
+
+Then deploy the program by running the `solana program deploy` command.
+
+```sh
+solana program deploy target/deploy/<your_program_name>.so
+```
+
+Upon successful deployment, you'll receive a Program ID. For example:
+
+```sh
+Program Id: AzKatnACpNwQxWRs2YyPovsGhgsYVBiTmC3TL4t72eJW
+```
+
+If you encounter an "insufficient funds" error during deployment, you may need
+to add SOL to your deployment wallet. Use the Solana CLI to request an airdrop:
+
+```sh
+solana airdrop 2
+```
+
+After receiving the airdrop, attempt the deployment again.
+
+<Callout>
+
+Ensure your Solana CLI is configured for the correct network (`Localnet`,
+`devnet`, `testnet`, or `mainnet-beta`) before deploying or requesting airdrops.
+</Callout>
+
+If you encounter the following error during program deployment, it indicates
+that your program size needs to be extended:
+
+```sh
+Error: Deploying program failed: RPC response error -32002: Transaction simulation failed: Error processing Instruction 0: account data too small for instruction [3 log messages ]
+```
+
+To resolve this, if you're using Solana CLI version 1.18 or later, run the
+following command:
 
-Now we’re ready to build and deploy our program! You can build the program by
-running `cargo build-bpf` and then running the command that is returned, it
-should look something like `solana program deploy <PATH>`.
+```sh
+solana program extend PROGRAM_ID 20000 -u d -k KEYPAIR_FILE_PATH
+```
+
+Replace `PROGRAM_ID` and `KEYPAIR_FILE_PATH` with your own values. For example:
+
+```sh
+ solana program extend HMDRWmYvL2A9xVKZG8iA1ozxi4gMKiHQz9mFkURKrG4 20000 -u d -k ~/.config/solana/id.json
+```
```

Comment:
Nice addition. In future we might have a 'deploying troubleshooting' resource we can link to but we can leave this here for now.